modules/timestamp/timestamp.py

```diff
@@ -0,0 +1,22 @@
+import datetime
+
+class Timestamp: 
+    """
+    Adds the time stamp to a given input (either video frame or telemetry)
+
+    ...
+    Attributes
+    ----------
+    data : 
+    timestamp : datetime.datetime
+
+    Methods
+    -------
+    __init__(data, timestamp (optional, defaults to time of initialization)) 
+        sets the frame and timestamp
+
+    """
+    def __init__(self, data, timestamp = datetime.datetime.now()):
```

Comment:
Since this is a data structure, it might be helpful to have type hints. I haven't enforced it elsewhere in the codebase but for a data structure its pretty important to have type hints that tell us exactly what constitutes the data structure. 

If you're unfamiliar, python type hints work like this: ```def int_func(num: int)``` defines num as an integer paramter. For timetstamp, it just needs to be ```timestamp: datetime```, ```datetime``` being the imported datetime object. 

I'd make data a union type, a union between the dict type and the numpy array-like type, here's the [documentation](https://numpy.org/doc/stable/reference/typing.html#module-numpy.typing)